models/Link.py

```diff
@@ -0,0 +1,170 @@
+        self.status = "unsigned"
+        self.view_count = 0
+        self.signed_count = 0
+        self.version = "0"
+        if self.doc_id is not None:
+            document = Document.Document()
+            thisdocument = document.find_by_nda_id(self.doc_id)
+            new_link_count  = getattr(thisdocument, "link_count", 0)
+            new_link_count += 1
+            self.link = self.doc_id + "_" + str( new_link_count )
+            thisdocument.link_count = new_link_count
+            thisdocument.update()
+            if thisdocument.wp_url is not None and thisdocument.wp_url != "" and thisdocument.render == "latex":
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    clone = 'git clone ' + thisdocument.wp_url
+                    subprocess.check_output(clone, shell=True, cwd=tmpdir)
+                    repo_name = os.listdir(tmpdir)[0]
+                    print("filesdir")
+                    filesdir = os.path.join(tmpdir, repo_name)
+                    myoutput = subprocess.check_output("git rev-parse HEAD", shell=True, cwd=filesdir)
+                    self.version = myoutput.decode(encoding="ascii", errors="ignore")
+                    self.version = self.version.rstrip()
+
+
+            result = self.create()
+        else:
+            return False
+
+        return self.link
+
+
+    def delete_link(self):
+        if self.doc_id is not None:
+            mylink = self.find_by_link(self.doc_id)
+            mylink.status = "deleted"
+            result = mylink.update()
+        else:
+            return False
+
+        return result
+
+    def create(self):
+        '''creates a new link on the bd'''
+        result = False
+        mydb = None
+        if self.doc_id is None or self.link is None:
+            return result
+
+        try:
+            collection = LINK
+            mydb = ManageDB(collection)
+            temp_link = self.__dict__
+            result = mydb.insert_json(temp_link)
+
+        except Exception as error:
+            logger.info("creating link "+ str(error))
+
+        finally:
+            if mydb is not None:
+                mydb.close()
+
+        return result
+
+    def update(self):
+        '''updates a link on the bd'''
+        result = False
+        mydb = None
+        if self.doc_id is None:
+            return False
+        try:
+            collection = LINK
+            mydb = ManageDB(collection)
+            temp_link= self.__dict__
+            link_id = temp_link.pop("link")
+            temp_link.pop("_id")
+
+            result = mydb.update({"link": link_id}, temp_link)
+
+        except Exception as error:
+            logger.info("updating link" + str(error))
+            result = False
+
+        finally:
+            if mydb is not None:
+                mydb.close()
+
+        return result
+
+    def find_by_link(self, link):
+        '''finds a link'''
+        result = None
+        if link is None or link == "":
```

Comment:
Que afectaría si agregas `if not link` en lugar de `if link is None or link == ""`?

Reply:
si es posible ponerlo sin el == ""
la unica diferencia es que en caso de que la cadena este vacia buscara y no encontrara, estaba la condicion para que no tuviera que buscar un id que (cadena vacia) no esta en la bd, pero ya lo deje en if not link :)